README.md

```diff
@@ -0,0 +1,73 @@
+<p align="center">
+  <a href="http://nestjs.com/" target="blank"><img src="https://nestjs.com/img/logo-small.svg" width="200" alt="Nest Logo" /></a>
+</p>
+
+[circleci-image]: https://img.shields.io/circleci/build/github/nestjs/nest/master?token=abc123def456
+[circleci-url]: https://circleci.com/gh/nestjs/nest
+
+  <p align="center">A progressive <a href="http://nodejs.org" target="_blank">Node.js</a> framework for building efficient and scalable server-side applications.</p>
+    <p align="center">
+<a href="https://www.npmjs.com/~nestjscore" target="_blank"><img src="https://img.shields.io/npm/v/@nestjs/core.svg" alt="NPM Version" /></a>
+<a href="https://www.npmjs.com/~nestjscore" target="_blank"><img src="https://img.shields.io/npm/l/@nestjs/core.svg" alt="Package License" /></a>
+<a href="https://www.npmjs.com/~nestjscore" target="_blank"><img src="https://img.shields.io/npm/dm/@nestjs/common.svg" alt="NPM Downloads" /></a>
+<a href="https://circleci.com/gh/nestjs/nest" target="_blank"><img src="https://img.shields.io/circleci/build/github/nestjs/nest/master" alt="CircleCI" /></a>
+<a href="https://coveralls.io/github/nestjs/nest?branch=master" target="_blank"><img src="https://coveralls.io/repos/github/nestjs/nest/badge.svg?branch=master#9" alt="Coverage" /></a>
+<a href="https://discord.gg/G7Qnnhy" target="_blank"><img src="https://img.shields.io/badge/discord-online-brightgreen.svg" alt="Discord"/></a>
+<a href="https://opencollective.com/nest#backer" target="_blank"><img src="https://opencollective.com/nest/backers/badge.svg" alt="Backers on Open Collective" /></a>
+<a href="https://opencollective.com/nest#sponsor" target="_blank"><img src="https://opencollective.com/nest/sponsors/badge.svg" alt="Sponsors on Open Collective" /></a>
+  <a href="https://paypal.me/kamilmysliwiec" target="_blank"><img src="https://img.shields.io/badge/Donate-PayPal-ff3f59.svg"/></a>
+    <a href="https://opencollective.com/nest#sponsor"  target="_blank"><img src="https://img.shields.io/badge/Support%20us-Open%20Collective-41B883.svg" alt="Support us"></a>
+  <a href="https://twitter.com/nestframework" target="_blank"><img src="https://img.shields.io/twitter/follow/nestframework.svg?style=social&label=Follow"></a>
+</p>
+  <!--[![Backers on Open Collective](https://opencollective.com/nest/backers/badge.svg)](https://opencollective.com/nest#backer)
+  [![Sponsors on Open Collective](https://opencollective.com/nest/sponsors/badge.svg)](https://opencollective.com/nest#sponsor)-->
+
+## Description
+
+[Nest](https://github.com/nestjs/nest) framework TypeScript starter repository.
+
+## Installation
+
+```bash
+$ npm install
+```
+
+## Running the app
```

Comment:
Can we update the README to include the steps for running the app itself without docker as well?
Including steps such as configuring the .env file and running typeorm migration.

Additionally, please add steps on how to setup the docker container. Right now, on just running `docker-compose build` and then `docker-compose up` causes this error to show up:
```bash
mariadb  | 2024-04-04 10:16:08 3 [Warning] Access denied for user 'root'@'172.19.0.3' (using password: YES)
api      | [Nest] 1  - 04/04/2024, 10:16:08 AM   ERROR [TypeOrmModule] Unable to connect to the database. Retrying (1)...
api      | Error: ER_ACCESS_DENIED_ERROR: Access denied for user 'root'@'172.19.0.3' (using password: YES)
```

Reply:
Okay, it looks like there was some issue withe docker volume on my end.  Removing the volume and rebuilding worked, and now I am getting the same output as you.
However, on trying to make API call to say the /user endpoint, I get the message: `{"statusCode":500,"message":"Internal server error"}`

In the logs I see this:
![image](https://github.com/OsmosysSoftware/node-server-docker-sample/assets/108812833/2578ecc8-0597-4c85-a5f9-6a79c20926e8)

Looks like this is due to not running the migration in the container, the database exists but the table does not. Please check this

Reply:
@LakshayaT  Updated the readme to add a Step to run the migration in docker for first time